File: 24-25-Feb/b1.py
def check(x, y):
    cnt = 0
    cnt+=(grid[x][y] == '#')+(grid[x][N - 1 - y] == '#')+(grid[N - 1 - x][y] == '#')+(grid[N - 1 - x][N - 1 - y] == '#')
    return min(cnt, 4 - cnt)

# 输入读取
N, U = map(int, input().split())
grid = [list(input().strip()) for _ in range(N)]

ans = 0
for i in range(N):
    for j in range(N):
        ans += check(i, j)
ans //= 4  # 每组被重复计算了4次

print(ans)

for _ in range(U):
    x, y = map(int, input().split())
    x -= 1
    y -= 1
    before = check(x, y)
    grid[x][y] = '.' if grid[x][y] == '#' else '#'
    after = check(x, y)
    ans += after - before
    print(ans)

File: 24-25-Feb/b1.cpp
#include<iostream>
using namespace std;
const int maxn=2025;
char map[maxn][maxn];
int n,m,ans;
int check(int x,int y){
    int cnt=0;
    cnt+=(map[x][y]=='#')+(map[x][n-1-y]=='#')+(map[n-1-x][y]=='#')+(map[n-1-x][n-1-y]=='#');
    return min(cnt,4-cnt);
}

int main(){
    cin>>n>>m;
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            cin>>map[i][j];
        }
    }
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            ans+=check(i,j);
        }
    }
    ans/=4;
    cout<<ans<<endl;
    while(m--){
        int x,y;
        cin>>x>>y;
        x--;
        y--;
        int before=check(x,y);
        if(map[x][y]=='#'){
            map[x][y]='.';
        }else{
            map[x][y]='#';
        }
        int after=check(x,y);
        ans+=after-before;
        cout<<ans<<endl;
    }
    return 0;
}